Updates for os_config

LazyDBA247-Anyvision opened the pull request:
1. remove k3s optimizations (not needed)
2. update memory limits (same as in gravity)
3. add network (arp cache) increase (Menards bug)

LazyDBA247-Anyvision committed:
OS_CONFIG: remove k3s optimizations

some scripts are for k3s not relevant to docker-compose

LazyDBA247-Anyvision committed:
OS_CONFIG: increase arp cache

found issue in Menards, this fixed the issue,
maybe? not the real source, but won't break anything...

ghost commented:
@sashas-anyvision, you can marge yoni commits, please?

sashas-anyvision reviewed (changes requested):
why are those changes done ? can we please specify in commit message what problems did it solve ?
otherwise we will find ourselves asking this question in several month and noone will remember why

sashas-anyvision commented on ansible/roles/os_config/tasks/network_optimization.yml:
```diff
@@ -0,0 +1,15 @@
+- name: update sysctl net.ipv4.neigh.default.gc_thresh1
```
i`m not sure this needs to be done .. are there any downsides to increasing this ?

OriBenHur-Any replied:
@LazyDBA247-Anyvision.

LazyDBA247-Anyvision replied:
@sashas-anyvision @OriBenHur-Any 
we not sure we saw this bug again, or if that was the main cause, but that was the setting i changed on a not working system, and 5 seconds after v2+insights got to healthy/working.
but i am sure the change won't break anything, so "better safe than sorry?"

sashas-anyvision commented:
@LazyDBA247-Anyvision  what about this PR?

sashas-anyvision commented on ansible/roles/os_config/tasks/memory_optimization.yml:
```diff
@@ -3,28 +3,28 @@
     limit_type: soft
     limit_item: nofile
-    value: 200000
+    value: 1000000
```
please describe when we discovered the problem .. what was the problem ..

LazyDBA247-Anyvision replied:
not aligned to our gravity installation and k8s installs settings.

sashas-anyvision commented on ansible/roles/os_config/tasks/memory_optimization.yml:
```diff
@@ -136,57 +136,28 @@
 - name: update nofile hard 1000
   pam_limits:
-    domain: '1000'
+    domain: '*'
```
please split into 2 commuits .. 
1) handles the "user" issue i.e. 1000 -> "*" and b.t.w. this is not related to compose gravity .. it was just a "luck" that we had only single user that run the compose :) if we would create 2 users on server it wont work .. just first user is 1000 
2) limits with problem statement ..

sashas-anyvision replied:
@LazyDBA247-Anyvision  ping ?

sashas-anyvision reviewed (changes requested):
great job . please see some comments

sashas-anyvision commented:
@LazyDBA247-Anyvision last call to fix the PR .. if not i`m closing it

LazyDBA247-Anyvision committed:
security.limits: root limit

root user is not affected from setting with *
setting the same limits as gravity/k8s installation does.
NOTE: group and wildcard limits are not applied to the root user. To set a limit for the root user, this field must contain the literal username root.

LazyDBA247-Anyvision committed:
sysctl: system wide open files

setting the same limits as gravity/k8s installation does.
2M for open files

LazyDBA247-Anyvision committed:
security.limits: all users

setting the same limits as gravity/k8s installation does.
once we had it on user 1000 (first) because memsql uid is 1000,
we decided to support on any uid/gid   mongo/gravity/etc...

sashas-anyvision commented:
@LazyDBA247-Anyvision  ping X2?

LazyDBA247-Anyvision commented:
@sashas-anyvision i fixed and commited all the requested changes?
what is missing?

sashas-anyvision commented:
approved .. yalla .. push the merge
[image: Employee profile]
Sasha Solganik
Backend Team Leader
Development Team
M +972 543348368 <+972%20543348368>
E alexanders@anyvision.co
W www.anyvision.co


On Sun, Nov 17, 2019 at 11:55 PM Yoni Nakache <notifications@github.com>
wrote:

> @sashas-anyvision <https://github.com/sashas-anyvision> i fixed and
> commited all the requested changes?
> what is missing?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/AnyVisionltd/compose-oneliner/pull/10?email_source=notifications&email_token=AL2OXVO3ATUGTF7I5QTOHG3QUG4U3A5CNFSM4I53OVA2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEEIXGHQ#issuecomment-554791710>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AL2OXVJACD4CRCZ7U45Y5JDQUG4U3ANCNFSM4I53OVAQ>
> .
>
